pkg/resource/deploy/source_eval.go

```diff
@@ -1502,12 +1510,34 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 		PluginChecksums:         req.GetPluginChecksums(),
 	}
 
-	// Before we calculate anything else run the transformations. First run the transforms for this resource,
-	// then it's parents etc etc
-	transforms, err := slice.MapError(req.Transforms, rm.wrapTransformCallback)
+	// This might be a resource registation for a resource that another process requested to be constructed. If so we'll
+	// have saved the pending transforms for this and we should use those rather than what is on the request.
+	var transforms []TransformFunction
+	pendingKey := fmt.Sprintf("%s::%s::%s", parent, t, name)
+	err = func() error {
+		rm.pendingTransformsLock.Lock()
+		defer rm.pendingTransformsLock.Unlock()
+
+		if pending, ok := rm.pendingTransforms[pendingKey]; ok {
+			transforms = pending
+		} else {
```

Comment:
What happens if the program specified transforms for the remote component _and_ then the implementation of the remote component (inside `Construct`) also adds more transforms to its request?

Reply:
Currently we ignore the transforms set inside the component. We could append them to the pending list, but need to be careful about duplication and ordering.